### ISSUE TITLE
[Feature] Add Print Controls

### ISSUE THREAD

**Issue by WolfwithSword** [greghesp/ha-bambulab#13]
Another port from the NR flow would be print control buttons.

Keep in mind that ideally they would only ever be active/available or do any work when the printer is in either a RUNNING or PAUSE state. Resume should only work when in a PAUSE, Pause only in RUNNING, Cancel/Stop in either one.

Here are the publish commands for each:

```
pause = { "print": { "sequence_id": "2008", "command": "pause" }, "user_id": "123456789" }

resume = { "print": { "sequence_id": "2009", "command": "resume"}, "user_id": "123456789" }

stop = { "print": { "sequence_id": "2010", "command": "stop" }, "user_id": "123456789" }
```

**Comment by WolfwithSword** [comment]
Definitely just a button platform. Could do some trickery with availability based on current states, or just block the calls in code depending on the states to be easy at first.

**Comment by AdrianGarside** [comment]
'Current stage' isn't currently being populated with my P1P. I don't see stg_cur in any of the mqtt payload I've captured.

**Comment by AdrianGarside** [comment]
gcode_state I see going  from IDLE -> PREPARE -> RUNNING but pausing the print didn't update it.

The pause caused this:
2023-02-11 16:23:30.077 DEBUG (Thread-6 (_thread_main)) [custom_components.bambu_lab.pybambu] b'{"mc_print":{"param":"[LINK] pause_print():Pause Print!reason= 0x8001","command":"push_info","sequence_id":"2740"}}'
2023-02-11 16:23:30.079 DEBUG (Thread-6 (_thread_main)) [custom_components.bambu_lab.pybambu] b'{"mc_print":{"param":"[LINK] Print state change to 0x03","command":"push_info","sequence_id":"2741"}}'
2023-02-11 16:23:33.359 DEBUG (Thread-6 (_thread_main)) [custom_components.bambu_lab.pybambu] b'{"mc_print":{"param":"[LINK] Resume Print!","command":"push_info","sequence_id":"2760"}}'
2023-02-11 16:23:33.364 DEBUG (Thread-6 (_thread_main)) [custom_components.bambu_lab.pybambu] b'{"mc_print":{"param":"[LINK] Print state change to 0x02","command":"push_info","sequence_id":"2762"}}'

**Comment by WolfwithSword** [comment]
We wouldn't need to use Currant Stage or any stage data for that verification. We can use the print_status / gcode_state. Its values are IDLE, RUNNING, PAUSE, FINISH, FAILED. After a print finishes, it stays stuck on finish or failed. Technically also PREPARE exists but this is so quick that it never really registers even on the X1C. 

Hmm, I'm not sure what those messages are. It looks like it paused the print, changed to state 0x03 which might be pause? Then 3 seconds later ran resume print? Or was that you?

Give it some time after pausing if it's in RUNNING, it may not change the state fully until it does the whole nozzle wipe part sometimes.

**Comment by AdrianGarside** [comment]
Yes that was me pausing the print briefly. I didn’t leave it paused long as I was worried about messing up the print quality with a gap or blob as I’ve seen in the past.
